librz/core/cfile.c

```diff
@@ -892,32 +892,58 @@ typedef struct {
 	RzBin *bin;
 } RzCoreLinkData;
 
-static bool linkcb(void *user, void *data, ut32 id) {
-	RzCoreLinkData *ld = user;
-	RzIODesc *desc = (RzIODesc *)data;
-
+static bool resolve_import_cb(RzCoreLinkData *ld, RzIODesc *desc, ut32 id) {
 	RzBinFile *bf = rz_bin_file_find_by_fd(ld->bin, desc->fd);
-	if (bf) {
-		RzListIter *iter;
-		RzBinSymbol *sym;
-		RzList *symbols = rz_bin_file_get_symbols(bf);
-		rz_list_foreach (symbols, iter, sym) {
-			if (!strcmp(sym->name, ld->name)) {
-				ld->addr = sym->vaddr;
-				return false;
-			}
+	if (!bf) {
+		return true;
+	}
+	RzListIter *iter;
+	RzBinSymbol *sym;
+	RzList *symbols = rz_bin_file_get_symbols(bf);
+	rz_list_foreach (symbols, iter, sym) {
+		if (!strcmp(sym->name, ld->name)) {
+			ld->addr = sym->vaddr;
+			return false;
 		}
 	}
 	return true;
 }
 
+static bool map_multi_dex(RzCore *core, RzIODesc *desc, ut32 id) {
+	if (!rz_str_endswith(desc->name, ".dex")) {
+		return true;
+	}
+
+	ut64 size = rz_io_desc_size(desc);
+	ut64 baddr = rz_io_map_next_available(core->io, 0x200000, size, 0x100000);
+	RZ_LOG_INFO("Mapping %s at 0x%" PFMT64x " with size 0x%" PFMT64x "\n", desc->name, baddr, size);
+	if (baddr != UT64_MAX) {
+		RzCoreFile *cf = rz_core_file_cur(core);
+		rz_io_use_fd(core->io, desc->fd);
+		RzBinOptions opt;
+		rz_core_bin_options_init(core, &opt, desc->fd, baddr, 0);
+		opt.xtr_idx = 0;
+		RzBinFile *binfile = rz_bin_open_io(core->bin, &opt);
+		if (!binfile) {
+			RZ_LOG_ERROR("Cannot load bin file %s.\n", desc->name);
+			return true;
+		}
+
+		rz_pvector_push(&cf->binfiles, binfile);
+		rz_core_bin_apply_all_info(core, binfile);
+		rz_core_cmd0(core, "\"(fix-dex,wx `ph sha1 $s-32 @32` @12 ; wx `ph adler32 $s-12 @12` @8)\"\n");
```

Comment:
At least some of that command could be converted into the API - `ph` and `wx` for sure.

Reply:
tbh, the whole cfile needs to be rewritten. i would keep this like this for now

Reply:
i will rewrite the whole this later